packages/react-theme-provider/src/components/ThemeProvider/useThemeProvider.ts

```diff
@@ -34,9 +33,10 @@ export const useThemeProvider = (
   );
 
   const parentTheme = useTheme();
-  const localTheme = state.theme;
+  const localTheme = draftState.theme;
+
+  const state = assign(draftState, { theme: mergeThemes(parentTheme, localTheme) });
```

Comment:
This is kind of weird since `assign` returns a reference to draftstate, you're pretty much doing

```ts
const state = draftState
```

Is this the expected behavior?

Reply:
This does a type safe transform from ThemeProviderDraftState (where `theme` is optional and could be `PartialTheme`) to ThemeProviderState (where `theme` is not optional). This allows compile time checking that the `state` object is a fully-formed `ThemeProviderState`.

It is equivalent to the following, but adds type safety that you didn't forget to assign the theme prop (or assigned it to something invalid):
```typescript
const state = draftState as ThemeProviderState;
state.theme = mergeThemes(...);
```